### PR TITLE
vucm: resolve mdns hostnames from code

### PR DESCRIPTION
**Pull request by rnovatorov**
[How can I find the IP address of a host using mdns](https://stackoverflow.com/a/35853322):

> According to the mdns specification (RFC 6762 §6.7) if the query received by an MDNSresponder comes from a source port other than 5353, this is an indication that the querier is a simple resolver, and the responder MUST send a UDP response directly back to the querier, via unicast, to the query packet's source IP address and port. It is due to this that the code above (and nslookup itself) works. And it works only when the responder abides by this stipulation, avahi does and the code works. But simpler implementations (like the esp8266 one) do not, so the code does not work either.

Excerpt from the RFC:

> [6.7](https://www.rfc-editor.org/rfc/rfc6762#section-6.7).  Legacy Unicast Responses
> 
> If the source UDP port in a received Multicast DNS query is not port
   5353, this indicates that the querier originating the query is a
   simple resolver such as described in [Section 5.1](https://www.rfc-editor.org/rfc/rfc6762#section-5.1), "One-Shot Multicast
   DNS Queries", which does not fully implement all of Multicast DNS.
   In this case, the Multicast DNS responder MUST send a UDP response
   directly back to the querier, via unicast, to the query packet's
   source IP address and port.  This unicast response MUST be a
   conventional unicast response as would be generated by a conventional
   Unicast DNS server; for example, it MUST repeat the query ID and the
   question given in the query message.  In addition, the cache-flush
   bit described in [Section 10.2](https://www.rfc-editor.org/rfc/rfc6762#section-10.2), "Announcements to Flush Outdated Cache
   Entries", MUST NOT be set in legacy unicast responses.

### CODE REVIEW

**Comment by rnovatorov** [comment]
Closes #6.

File: enapter/__init__.py [hunk 1]
```diff
@@ -1,11 +1,12 @@
 __version__ = "0.3.0"
```
Comment (kulti) [inline enapter/__init__.py:1]: Еще версию поднять нужно.

File: enapter/vucm/app.py [hunk 2]
```diff
@@ -21,8 +21,14 @@ async def _run(self):
         logger = log.new(level=self._config.log_level)
         tasks = set()
 
+        mdns_resolver = None
+        if self._config.mqtt.host.endswith(".local"):
+            mdns_resolver = mdns.Resolver()
```
Comment (kulti) [inline enapter/vucm/app.py:26]: Что-то размазалась логика по двум компонентам.

Если .local, то надо резолвить имя через mdns, иначе никак (в клиенте проверка и эксепшен, если резолвера не будет). Почему бы вместо того, чтобы передавать резолвер в клиент, создавать резолвер в клиенте и резолвить хост? Или можно тут резолвить и передавать готовый ip, но это чуть хуже по слоям абстракции, имхо.

**Review by kulti — APPROVED** [review]
Looks good. A version bump is required.

File: enapter/vucm/app.py (hunk 2 above)
Comment (rnovatorov) [reply enapter/vucm/app.py]: Да, согласен, немного странно получилось.

Идея была в том, чтобы оставить задел на юнит-тесты, позволив передавать мок резолвера снаружи. Переделал сейчас через фабрику - теперь можно будет и тесты сделать, и в другой компонент логика больше не утекает.

File: enapter/__init__.py (hunk 1 above)
Comment (rnovatorov) [reply enapter/__init__.py]: Поднял до 0.4.0.

**Pull request merged**